### PR TITLE
Add Notebook, demonstrating saving and uploading of evaluators.

### PR DESCRIPTION
**Pull request by nick863**
# Description

In this PR we are adding the notebook, demonstrating saving and uploading of the standard evaluator to Azure workspace or registry. At the end of the notebook we are introducing function, uploading all the evaluators in the given namespace.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [x] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-evals/samples/EvaluatorUpload.ipynb [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import json\n",
+    "import pandas as pd\n",
+    "import shutil\n",
+    "import uuid\n",
+    "import yaml\n",
+    "\n",
+    "from azure.ai.ml._ml_client import MLClient\n",
```
Comment (nagkumar91) [inline src/promptflow-evals/samples/EvaluatorUpload.ipynb:28]:
```suggestion
    "from azure.ai.ml import MLClient\n",
```

Comment (nick863) [reply src/promptflow-evals/samples/EvaluatorUpload.ipynb]: Fixed

**Review by nagkumar91 — DISMISSED** [review]

File: src/promptflow-evals/samples/EvaluatorUpload.ipynb [hunk 2]
```diff
@@ -0,0 +1,497 @@
...
+    "{\r\n",
+    "    \"resource_group_name\": \"resource-group-name\",\r\n",
+    "    \"workspace_name\": \"ws-name\",\r\n",
+    "    \"subscription_id\": \"subscription-uuid\",\r\n",
+    "    \"registry_name\": \"registry-name\"\r\n",
+    "}\r\n",
+    "```\"\r\n",
+    "\n",
+    "\n",
+    "``````"
```
Comment (ninghu) [inline src/promptflow-evals/samples/EvaluatorUpload.ipynb:229]: nit: this line is probably not needed.

File: src/promptflow-evals/samples/EvaluatorUpload.ipynb [hunk 3]
```diff
@@ -0,0 +1,497 @@
...
+   "cell_type": "markdown",
+   "id": "4a94aab5-73f8-4c7d-a7e0-a92853db0198",
+   "metadata": {},
+   "source": [
+    "Now we have all the tools to upload our model to Azure\n",
+    "### Uploading data to Azure\n",
+    "First we will need to authenticate to azure. For this purpose we will use the the configuration file of the net structure.\n",
+    "```json\r\n",
+    "{\r\n",
+    "    \"resource_group_name\": \"resource-group-name\",\r\n",
```
Comment (ninghu) [inline src/promptflow-evals/samples/EvaluatorUpload.ipynb:221]: what would happen if use both `registry_name` and `workkspace_name` to initialize the `ml_client` and call `ml_client.evaluators.create_or_update(eval)`.

File: src/promptflow-evals/samples/EvaluatorUpload.ipynb [hunk 4]
```diff
@@ -0,0 +1,497 @@
...
+    ")\n",
+    "pf.stream(run)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6e7ee69f-4db3-4f5f-874a-5dc95d2e2f7c",
+   "metadata": {},
+   "source": [
+    "<font color='red'>**Hack for standard evaluators.**</font> If we will try to load our evaluator directly, we will get an error, because we try to modify `__path__` variable, which will work inside the python package, however, it will fail if we will try to run evaluator as a standalone script. To fix it, we will remove this line from our code. **This code is not needed for custom user evaluators!**"
```
Comment (ninghu) [inline src/promptflow-evals/samples/EvaluatorUpload.ipynb:139]: so if we can remove  `__path__ = __import__("pkgutil").extend_path(__path__, __name__)`  from `__init__.py`, we don't need below workaround, correct?

File: src/promptflow-evals/samples/EvaluatorUpload.ipynb [hunk 5]
```diff
@@ -0,0 +1,497 @@
...
+   "source": [
+    "import os\n",
+    "import json\n",
+    "import pandas as pd\n",
+    "import shutil\n",
+    "import uuid\n",
+    "import yaml\n",
+    "\n",
+    "from azure.ai.ml import MLClient\n",
+    "from azure.identity._credentials.default import AzureCliCredential\n",
```
Comment (ninghu) [inline src/promptflow-evals/samples/EvaluatorUpload.ipynb:29]: Suggest change to use DefaultAzureCredential

`from azure.identity import DefaultAzureCredential`

File: src/promptflow-evals/samples/EvaluatorUpload.ipynb [hunk 6]
```diff
@@ -0,0 +1,497 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "36338014-05f9-4f37-9fb0-726bb1c137b8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "config_ws = configuration.copy()\n",
+    "del config_ws[\"registry_name\"]\n",
+    "\n",
+    "credential = AzureCliCredential()\n",
```
Comment (ninghu) [inline src/promptflow-evals/samples/EvaluatorUpload.ipynb:262]: Same here

File: src/promptflow-evals/samples/EvaluatorUpload.ipynb (hunk 3 above)
Comment (nick863) [reply src/promptflow-evals/samples/EvaluatorUpload.ipynb:221]: In this case, the validation error will be raised:
```
--> 166     raise ValidationException(
    167         message="Both workspace_name and registry_name cannot be used together, for the ml_client.",
    168         no_personal_data_message="Both workspace_name and registry_name are used for ml_client.",
    169         target=ErrorTarget.GENERAL,
    170         error_category=ErrorCategory.USER_ERROR,
    171     )
```

File: src/promptflow-evals/samples/EvaluatorUpload.ipynb (hunk 2 above)
Comment (nick863) [reply src/promptflow-evals/samples/EvaluatorUpload.ipynb]: Fixed.

File: src/promptflow-evals/samples/EvaluatorUpload.ipynb (hunk 5 above)
Comment (nick863) [reply src/promptflow-evals/samples/EvaluatorUpload.ipynb]: Fixed

**Review by nagkumar91 — DISMISSED** [review]

File: src/promptflow-evals/samples/EvaluatorUpload.ipynb [hunk 7]
```diff
@@ -0,0 +1,494 @@
...
+   "source": [
+    "import os\n",
+    "import json\n",
+    "import pandas as pd\n",
+    "import shutil\n",
+    "import uuid\n",
+    "import yaml\n",
+    "\n",
+    "from azure.ai.ml import MLClient\n",
+    "from azure.identity._credentials.default import DefaultAzureCredential\n",
```
Comment (ninghu) [inline src/promptflow-evals/samples/EvaluatorUpload.ipynb:29]: why not "from azure.identity import DefaultAzureCredential"?

Comment (nick863) [reply src/promptflow-evals/samples/EvaluatorUpload.ipynb]: Fixed

**Review by ninghu — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**